lib/services/front.ts

```diff
@@ -0,0 +1,349 @@
+     */
+    protected activateMessageListener(): void {
+        // This swallows response attempts to the channel, since we notice them on the inbox instead
+        MessageService.app.post('/front-dev-null', (_formData, response) => {
+            response.send();
+        });
+        // Create an endpoint for this listener and enqueue events
+        MessageService.app.post(`/${this.serviceName}/`, (formData, response) => {
+            this.queueEvent({
+                data: {
+                    cookedEvent: {
+                        context: formData.body.conversation.id,
+                        type: 'event',
+                    },
+                    rawEvent: formData.body,
+                    source: this.serviceName,
+                },
+                workerMethod: this.handleEvent,
+            });
+            response.send();
+        });
+    }
+
+    /**
+     * Deliver the payload to the service. Sourcing the relevant context has already been performed
+     * @param data - The object to be delivered to the service
+     * @returns {Promise<MessengerEmitResponse>} - Response from the service endpoint
+     */
+    protected sendPayload(data: FrontCommentEmitContext|FrontConversationEmitContext): Promise<MessengerEmitResponse> {
+        if (data.type === 'comment') {
+            // The event is sent to the comment method
+            return FrontService.session.comment.create(data).then((comment: Comment) => {
+                return {
+                    response: {
+                        message: comment.id,
+                        thread: data.conversation_id,
+                    },
+                    source: this.serviceName,
+                };
+            });
+        } else if (data.type === 'message') {
+            // The event is sent to the message (reply) method
+            return FrontService.session.message.reply(data).then(() => {
+                return {
+                    response: {
+                        message: `${data.author_id}:${new Date().getTime()}`,
+                        thread: data.conversation_id,
+                    },
+                    source: this.serviceName,
+                };
+            });
+        } else if (data.type === 'conversation') {
+            // The event is sent to the message (create) method
+            return FrontService.session.message.send(data).then(() => {
+                // This is because the response is ASAP and doesn't include the conversation id
+                return this.findConversation(data.subject)
+                .then((conversationId) => {
+                    return {
+                        response: {
+                            message: `${data.author_id}:${new Date().getTime()}`,
+                            thread: conversationId,
+                            url: `https://app.frontapp.com/open/${conversationId}`,
+                        },
+                        source: this.serviceName,
```

Comment:
serviceName is a static variable, it being a property of the filename, that is exposed as a object property via the get method, which this is using.